### PR TITLE
Reduce test warnings

### PR DESCRIPTION
**Pull request by Yash-10**
In reference to #1074, this PR aims to reduce test warnings. Some warnings might need further treatment to mitigate/suppress them but this implementation fixes some of the obvious ones.

Thanks!

### CODE REVIEW

**Comment by Yash-10** [comment]
I am sorry for a bunch of small pushes..

File: tests/tests_plotting/test_static.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def test_plot_trajectory_sets_label():
 
     op = StaticOrbitPlotter()
     trajectory = churi.sample()
-    op.plot_body_orbit(Mars, J2000, label="Mars")
+    op.plot_body_orbit(Mars, Time(J2000, scale="tdb"), label="Mars")
```
Comment (astrojuanlu) [inline tests/tests_plotting/test_static.py:65]: Let's use `J2000_TDB` instead https://github.com/poliastro/poliastro/blob/a43006b0cf6b9671128ff4e4ed82f45bea6d5205/src/poliastro/constants/general.py#L104-L106

Comment (astrojuanlu) [inline tests/tests_plotting/test_static.py:65]: Let's use `J2000_TDB` instead https://github.com/poliastro/poliastro/blob/a43006b0cf6b9671128ff4e4ed82f45bea6d5205/src/poliastro/constants/general.py#L104-L106

File: tests/tests_twobody/test_orbit.py [hunk 2]
```diff
@@ -173,7 +173,9 @@ def test_apply_maneuver_changes_epoch():
 def test_orbit_from_ephem_with_no_epoch_is_today():
     # This is not that obvious http://stackoverflow.com/q/6407362/554319
     body = Earth
-    ss = Orbit.from_body_ephem(body)
+    epoch = Time.now().tdb
+    ephem = Ephem.from_body(body, epoch)
+    ss = Orbit.from_ephem(Sun, ephem, epoch)
```
Comment (astrojuanlu) [inline tests/tests_twobody/test_orbit.py:178]: The tests of `Orbit.from_body_ephem` require further discussion.

If we are still using it internally, even if it's deprecated, it should be tested. And if it has to be tested, then we should silence the deprecation warnings.

But if we are not using it internally anymore, then this approach is on the right path, but I would try not to call `Ephem.from_body` (and use a custom, dummy `Ephem`).

File: tests/tests_twobody/test_orbit.py [hunk 3]
```diff
@@ -1173,6 +1178,7 @@ def test_orbit_change_attractor_out_of_SOI():
     assert "ValueError: Orbit is out of new attractor's SOI" in excinfo.exconly()
 
 
+@pytest.mark.filterwarnings("ignore::poliastro.twobody.orbit.PatchedConicsWarning")
```
Comment (astrojuanlu) [inline tests/tests_twobody/test_orbit.py:1181]: Instead of filtering the warning, should we assert that it's emitted?

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
Thanks a lot @Yash-10 ! This is on the right path. Some fixes are good as they are and I didn't say anything, for some others I left a few comments.

**Review by astrojuanlu — APPROVED** [review]
Looks good to me!

**Comment by astrojuanlu** [comment]
As you say, there are still some warnings left but this is strictly better than before. Thanks a lot!

**Comment by Yash-10** [comment]
Thanks @astrojuanlu! Yes, those warnings which include `ERFA`, `Orbit.pqw` would need further treatment but can be dealt in the future too!

Thanks!!

**Pull request merged**